### ISSUE TITLE
Sync/Mirror Issues and Pull-Requests on migrated repository

### ISSUE THREAD

**Issue by M4NH3X** [go-gitea/gitea#18369]
### Feature Description

**Feature:** Want to sync/mirror Issue/PR on demand or timely interval.
**Current situation:** When migration (remote to gitea) is done and items like Issues/PR are checked, they are pulled perfectly initially. However, it's one-off and there is no way to fetch new issues/PR later on.
**Workaround:** One has to once again initiate a new migration whenever they need to pull new issues/PR which is not idle and efficient. Another solution would be to use Mirror feature, but that only allows branches, tags, and commits.
**Why:** It will help me to be in sync with my remote repository all the time and if in future I want to go Gitea exclusive then it can be done easily.

Version: Gitea Version: 1.15.10
PS: You guys are amazing btw and thank you for all the work and and providing this amazing project for free. :heart:

**Comment by ObiWanCannobi** [comment]
Being able to sync pull requests and issues would make gitea the perfect alternative to github for me, personally.  +1

**Comment by rekby** [comment]
It need to full backup of repository.

Issues can explain decisions, some regression test in sources contains issues number and if issues lost - it lost some information about fixed error.

**Comment by Benjamin-Loison** [comment]
Note that releases have the same sync issue.

**Comment by hramrach** [comment]
Is only github mirroring supported?

**Comment by lunny** [comment]
Please follow #20311

**Comment by hramrach** [comment]
That's exactly it,  #20311 is only addressing github.

**Comment by davidlovas** [comment]
I have been re-deriving this feature (#20311) against current `main` and testing it at scale, and I'd like to bring it upstream. Posting here since this is the proposal home.

The mirror lets a pull mirror optionally sync issues and PRs, along with their comments, reviews, timeline events, labels, milestones and reactions, as a read-only copy. Everything renders natively and every write is refused, and it stays current on the normal pull-mirror cycle using an incremental `updated` watermark rather than the Search API.

I have stress tested this with a mirror of the gitea repo itself, which is live and browsable here: https://gitea.rexovas.com/rexovas/gitea

The main changes from the original branch, mostly from problems that only appear at scale:

- Re-sync is now idempotent: issues, PRs, comments and reviews store their remote id (the "unique external id" @lunny raised on #20311) and upsert on it, which resolves the `issue_index` duplicate-key error @ajvpot reported on subsequent syncs.
- The read-only display shows everything but locks writes. It is gated on a per-mirror flag (`IsMirrorWithMetadata`) rather than the coarse `!IsMirror` check, so existing plain mirrors are unaffected and a blocked write returns a clean message instead of a 500.
- The backfill survives rate-limit windows and resumes cleanly, which is what let the go-gitea import finish unattended.

One thing became clear early: at this scale the REST migration path is not viable, since every comment, review, reaction and timeline event is a separate call and more tokens only raise the budget rather than reduce the number of requests. I have therefore implemented a GraphQL downloader that fetches an issue together with its comments, reactions and timeline in a single request. It is useful on its own for any GitHub migration, so I have opened it as a standalone PR (#38694, which closes #13243) to land first, with the mirror building on top.

For sequencing upstream I would suggest:

1. The GraphQL downloader and its retry transport (#38694 / #13243), standalone.
2. The read-only metadata mirror (this feature), as a series of small stacked PRs.
3. Timeline events (always synced) and reactions (opt-in) over GraphQL, both isolated so a failure in either cannot abort an import.

A few items I have deliberately deferred: reaction freshness (reactions do not bump `updated_at`, so keeping them current needs a periodic rescan), review-comment reactions (deep nesting hits GitHub's node limit), the assign and review-request events (they need an original-assignee field), recreating the source owner so the mirror can sit at the upstream path, a small migration-form cleanup, and releases.

Scope is GitHub-only, matching #20311, but it is built on the migration Downloader interface, so it can be extended to support other migration sources later. You can see the other three planned PRs in the fork here: https://github.com/davidlovas/gitea

cc @harryzcy
